src/libraries/Microsoft.Extensions.Logging/src/LoggerInformation.cs

```diff
@@ -44,22 +45,29 @@ public bool IsEnabled(LogLevel level)
 
     internal readonly struct ScopeLogger
     {
-        public ScopeLogger(ILogger logger, IExternalScopeProvider externalScopeProvider)
+        public ScopeLogger(ILogger? logger, IExternalScopeProvider? externalScopeProvider)
         {
+            if (logger == null && externalScopeProvider == null)
+            {
+                throw new ArgumentNullException(nameof(logger), SR.Error_LoggerRequiredWithoutExternalScopeProvider);
```

Comment:
This should just be a Debug.Assert.  Users should not be able to break this contract.